src/grandorgue/GOSoundEngine.cpp

```diff
@@ -576,27 +596,39 @@ void GOSoundEngine::CreateReleaseSampler(GO_SAMPLER* handle)
 					}
 				}
 			}
-			unsigned cross_fade_len = this_pipe->GetReleaseCrossfadeLength();
-			new_sampler->fader.NewAttacking(gain_target, cross_fade_len, m_SampleRate);
 
-			if (m_ReleaseLength > 0)
-			{
-				if ( m_ReleaseLength < gain_decay_length || gain_decay_length == 0 )
-					gain_decay_length = m_ReleaseLength;
-			}
-
-			if (gain_decay_length > 0)
-				new_sampler->fader.StartDecay(gain_decay_length, m_SampleRate);
 
-			if (m_ReleaseAlignmentEnabled && release_section->SupportsStreamAlignment())
-			{
-				release_section->InitAlignedStream(&new_sampler->stream, &handle->stream);
-			}
-			else
-			{
-				release_section->InitStream(&m_ResamplerCoefs, &new_sampler->stream, this_pipe->GetTuning() / (float)m_SampleRate);
-			}
-			new_sampler->is_release = true;
+            /*  Algorithm to Choose Which Release Truncation or Scaling Method to activate. */
+			unsigned cross_fade_len = this_pipe->GetReleaseCrossfadeLength();
+			unsigned truncation_fade_len = this_pipe->GetReleaseTruncationLength();
+			new_sampler->fader.NewAttacking(gain_target, cross_fade_len, m_SampleRate);
+			
+			/* If Release Length Set in GO GUI toolbar is larger than 0 */
+            if (m_ReleaseLength > 0)
+            {
+                /* If Gain Decay Length (gain_decay_length) is greater than toolbar value (m_ReleaseLength), or equal to 0, gain_decay_length = toolbar value. */
+                if ( m_ReleaseLength < gain_decay_length || gain_decay_length == 0 )
+                    gain_decay_length = m_ReleaseLength;
+            }
+
+            /* If truncation values exist and release scaling has not been enabled, gain_decay_length = truncation value. */
+            else if (truncation_fade_len > 0 && gain_decay_length == 0)
+            { 
+                gain_decay_length = truncation_fade_len;
+            }
+
+            if (gain_decay_length > 0)
```

Comment:
The lines from here have not been changed. Please leave them in the original state (maybe it is a tab stuff)